doc/docs/setting_up_application.md

```diff
@@ -1,27 +1,26 @@
+4. If the device has not been used with the nRF Connect Bluetooth Low Energy app before, you may be asked to update the J-Link firmware and connectivity firmware for the device. You need to have the correct connectivity firmware on the nRF SoC to continue. The programming can take up to 30 seconds.
 
-    - Show the device at the top of the list by making it favorite.
-    - Rename the device.
-    - View and select between device Com Ports.
+    ![Programming prompt](./screenshots/nrfconnect_ble_program_prompt.png "Programming prompt")
 
-The name of the selected device is displayed on the navigation bar.
+    !!! note "Note"
+          If programming fails, check the solution to the [Programming with J-Link does not work](./troubleshooting.md#programming-with-j-link-does-not-work) issue on the Troubleshooting page.
 
-If the device has not been used with the nRF Connect Bluetooth Low Energy app before, you may be asked to update the J-Link firmware and connectivity firmware for the device. You need to have the correct connectivity firmware on the nRF SoC to continue. When the nRF SoC has been programmed with the correct firmware, the nRF Connect Bluetooth Low Energy app proceeds to connect to it over Universal Asynchronous Receiver/Transmitter (UART). When the connection is established, the device appears in the main view.
-
-!!! note "Note"
-      When using OS X, an issue with the SEGGER J-Link OB firmware leads to the corruption of long packets over UART. See [www.nordicsemi.com/nRFConnectOSXfix](www.nordicsemi.com/nRFConnectOSXfix) for more information.
+When the nRF SoC has been programmed with the correct firmware, the nRF Connect Bluetooth Low Energy app proceeds to connect to it over Universal Asynchronous Receiver/Transmitter (UART). When the connection is established, the device appears in the main view.
 
 ![Local device view](./screenshots/nRF_connect_local_device.png "Local device view")
 
-The local device is labeled Adapter to signal that it is connected locally to the computer. The device name and Bluetooth device address are shown at the top. The attribute table of the device is shown below them.
+The local device is labeled Adapter to signal that it is connected locally to the computer. The device name and Bluetooth device address are shown at the top. The attribute table of the device is shown below them. Use the [**Device options**](./overview_and_ui.md#device-options) button to see actions and configurations available for the device.
 
-To expand a menu that shows actions and configurations available for the local device, click the **Device options** icon ![Device options icon](./screenshots/cog.png).
\ No newline at end of file
+!!! note "Note"
+      When using OS X, an issue with the SEGGER J-Link OB firmware leads to the corruption of long packets over UART. See [www.nordicsemi.com/nRFConnectOSXfix](https://devzone.nordicsemi.com/nordic/nordic-blog/b/blog/posts/nrf-connect-v10-release#osxissue) for more information.
```

Comment:
Use “macOS” instead of “OS X” here and in `troubleshooting.md` since it has been renamed years ago?